src/cr/cube/matrix/cubemeasure.py

```diff
@@ -87,851 +94,1251 @@ def _slice_idx_expr(cls, cube, slice_idx):
         return np.s_[slice_idx]
 
 
-# === MEANS ===
+# === COUNTS (UNWEIGHTED & WEIGHTED) ===
 
 
-class _BaseCubeMeans(_BaseCubeMeasure):
-    """Base class for mean cube-measure variants."""
+class _BaseCubeCounts(_BaseCubeMeasure):
+    """Base class for count (weighted & unweighted) cube-measure variants."""
 
-    def __init__(self, dimensions, means):
-        super(_BaseCubeMeans, self).__init__(dimensions)
-        self._means = means
+    def __init__(self, dimensions, counts, diff_nans):
+        super(_BaseCubeCounts, self).__init__(dimensions)
+        self._counts = counts
+        self._diff_nans = diff_nans
 
     @classmethod
-    def factory(cls, cube, dimensions, slice_idx):
-        """Return _BaseCubeMeans subclass instance appropriate to `cube`.
-
-        Raises `ValueError` if the cube-result does not include a cube-means measure.
-        """
-        if cube.means is None:
-            raise ValueError("cube-result does not contain cube-means measure")
-        dimension_types = cube.dimension_types[-2:]
-        CubeMeansCls = (
-            _MrXMrCubeMeans
-            if dimension_types == (DT.MR, DT.MR)
-            else _MrXCatCubeMeans
-            if dimension_types[0] == DT.MR
-            else _CatXMrCubeMeans
-            if dimension_types[1] == DT.MR
-            else _CatXCatCubeMeans
+    def factory(cls, weighted, cube, dimensions, slice_idx):
+        """Return _BaseCubeCounts subclass instance appropriate to `cube`
+
+        Chooses between unweighted and weighted counts based on `type`.
+        """
+        dimension_type_strings = tuple(
+            "MR"
```

Comment:
Strings felt gross - didn't like the existing enums because I want to be very specific about how I'm categorizing dimension types and none of the existing ones were good enough.